tests/core/flags/test_core_flags.odin

```diff
@@ -1113,6 +1113,13 @@ test_if_map_cstrings_get_freed :: proc(t: ^testing.T) {
 
 @(test)
 test_os_handle :: proc(t: ^testing.T) {
+	defer if !testing.failed(t) {
```

Comment:
Should be
```odin
defer {
    if !testing.failed(t) {
        ...
    }
}
```
right?

Currently it will check if the test failed at the beginning of the test, which is always `false`.

Reply:
I made a discussion for this since it seems like something that could be confusing. #3782

Reply:
Oh strange I could've sworn it was a thing used to defer something IF the condition is currently true.
Because
```odin
if true {
    defer blah()
}
```
Would execute blah at the end of that if block, not at the end of the wanted scope.

I really thought `defer if` was created to get around that, @gingerBill ?

Reply:
I did a quick peek through `git log` for `defer if` and all I could find was 118ab60588832904e6a980c712702f71b4f7ec15 by @Kelimion.

Reply:
I think I'm just mistaken and I have to look over some of my codebases for bugs :laughing:

Reply:
Seems pretty solid evidence that `defer if ...` is just shorthand for `defer { if ... }`, because it's used prodigiously for clean-up only iff errors are (not) present, as appropriate.

![image](https://github.com/odin-lang/Odin/assets/194614/0d84e6e7-6ba8-4f96-85bf-8ba390c1f548)

Reply:
And really that's no different than the difference between `if cond do single_statement()` and `if cond { single_statement() }`. Really a way to avoid opening two blocks on one line, i.e. `defer { if cond { do_stuff() } }`.

Reply:
Of note, `defer` works with blocks optionally in general. I.e. `defer a = true` works just fine, same as the common and idiomatic `defer delete(x)`. So I believe the `defer if` distinction might be human language mis-parsing on our parts.

Reply:
Right, if you want a conditional defer: `if defer`
Deferred conditional: `defer if`

Reply:
> Right, if you want a conditional defer: `if defer` Deferred conditional: `defer if`

`if defer` won't do what you want though, the if creates a new scope that the defer is ran after.

Reply:
> `if defer` won't do what you want though, the if creates a new scope that the defer is ran after.

Then `if defer` might be a thing the compiler should error on as being a pointless and confusing construct.

Reply:
But mainly I wrote it out that way because just by reading left to right we can rephrase the Odin code into English that highlights we want `defer if` (deferred conditional), not `if defer` (conditional defer) if we're unsure which of the two we should write.

That there's the confusion to begin with and that the 2nd phrasing is nonsensical in practice suggests it should be made an error.

Reply:
I can't get `if defer a { /* ... */ }` to compile, if this was meant as a concrete and not a hypothetical syntax example.

Reply:
Confirmed. Well, nothing to worry about then.

Reply:
I thought we all meant `if x do defer ...` here

Reply:
Yes, I tried `if a defer` and it doesn't compile.